snuba/datasets/dataset.py

```diff
@@ -123,6 +115,15 @@ def get_query_processors(self) -> Sequence[QueryProcessor]:
         These are applied in sequence in the same order as they are defined and are supposed
         to be stateless.
         """
+        return self._get_custom_query_processors() + [
+            PrewhereProcessor()
+        ]
+
+    def _get_custom_query_processors(self) -> Sequence[QueryProcessor]:
+        """
+        Returns the list of query processors that can be specified by the
+        dataset configuration
+        """
         return []
```

Comment:
I'm not sure this split is necessary — I think it's fine for datasets to just append or prepend their custom processors to the value returned by `super`.

Ultimately, I don't think this processor this should act on `snuba.query.query.Query` (note: that's a lot of `query`) at all, it should act on `snuba.clickhouse.query.Query` instead (since `{get,set}_prewhere` should be moved there as well — that is what I thought we were talking about [here](https://github.com/getsentry/snuba/pull/574/files#r344992759) in case it wasn't clear), so I'm not sure we're going to get enough long-term value out of this to justify splitting `get_query_processors` and `_get_custom_query_processors`.

Reply:
I saw two issues in the previous version we discussed about (https://github.com/getsentry/snuba/pull/574/files#r344992759):
1) make Prewhere processing automatic for all datasets
2) make Prewhere part of the Clickhouse query format. 

The first is achievable here (thus this method), while the second needs more work. But no strong opinion on this solution since there are only a handful of datasets and making prewhere mandatory is always going to be possible. Will move it back to the get_query_processors method.